lib/galaxy/web/form_builder.py

```diff
@@ -235,6 +236,27 @@ def get_html( self, prefix="" ):
         rval += '<div class="toolParamHelp">This Galaxy server allows you to upload files via FTP.  To upload some files, log in to the FTP server at <strong>%s</strong> using your Galaxy credentials (email address and password).</div>' % self.ftp_site
         return rval
 
+class GenomespaceFileField(BaseField):
+    """
+    A genomspace file browser field.
+    >>> print GenomespaceFileField( "foo", 100 ).get_html()
+    <input type="text" name="foo" ...>
+    """
+    def __init__(self, name, value=None):
+        self.name = name
+        self.value = value or ""
+
+    def get_html(self, prefix=""):
+        return unicodify('<script src="https://gsui.genomespace.org/jsui/upload/gsuploadwindow.js"></script>'
```

Comment:
When we originally started working on this, the tool parameters were still being rendered on the server side. Since it's now being rendered on the client-side, we've added a backbone component for it, but also left this bit as is since workflows don't fully support client-side rendering yet. Once that conversion is done, I hope that this bit can be nuked, preferably from orbit :-)